src/test/java/com/ericsson/ei/flowtests/FlowTest.java

```diff
@@ -163,35 +185,50 @@ public void flowTest() {
             }
 
             // wait for all events to be processed
-            int processedEvents = 0;
+            long processedEvents = 0;
+            long act = 0;
             while (processedEvents < eventsCount) {
                 String countStr = System.getProperty("eiffel.intelligence.processedEventsCount");
                 String waitingCountStr = System.getProperty("eiffel.intelligence.waitListEventsCount");
                 if (waitingCountStr == null)
                     waitingCountStr = "0";
                 Properties props = admin.getQueueProperties(queue.getName());
                 int messageCount = Integer.parseInt(props.get("QUEUE_MESSAGE_COUNT").toString());
+                act = countProcessedEvents(database, event_map);
+                log.info(act+" countStr="+ countStr + " waitingCountStr=" + waitingCountStr + "messageCount=" + messageCount +
+                        " " + (Integer.parseInt(countStr) - Integer.parseInt(waitingCountStr) - messageCount));
+
                 processedEvents = Integer.parseInt(countStr) - Integer.parseInt(waitingCountStr) - messageCount;
+                processedEvents = Math.min(processedEvents, act);
```

Comment:
Why do you still compute the processed events? Why not just assign act to processedEvents?